src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -39,6 +39,14 @@ import { relativeToAbsolute } from '../../nav_links/to_nav_link';
 export const isModifiedOrPrevented = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) =>
   event.metaKey || event.altKey || event.ctrlKey || event.shiftKey || event.defaultPrevented;
 
+// TODO: replace hard-coded values with a registration function, so that apps can control active nav links similar to breadcrumbs
+const aliasedApps: { [key: string]: string[] } = {
+  discover: ['data-explorer'],
+};
+
+export const isActiveNavLink = (appId: string | undefined, linkId: string): boolean =>
+  !!(appId === linkId || aliasedApps[linkId]?.includes(appId || ''));
```

Comment:
nit - I think `array.includes()` works the same with both `undefined` or `''`, so I'm not sure `appId || ''` is necessacry.

Reply:
@joshuarrrr Yes, you're right. Should I raise a PR for this?

Reply:
@joshuarrrr You're right,  both `undefined` and `''` works in this case. However, it could cause ts linting errors since the the type doesn't match : `Type 'undefined' is not assignable to type 'string'`. So I think we should leave it as is.